packages/core/logging/core-logging-server-internal/src/logger_adapter.ts

```diff
@@ -6,11 +6,16 @@
  * Side Public License, v 1.
  */
 
+import { has } from 'lodash';
+import { set } from '@elastic/safer-lodash-set';
 import { LogRecord, Logger, LogMeta } from '@kbn/logging';
 
+/** @internal */
+export type GlobalMeta = Map<string, unknown>;
```

Comment:
Yeah I didn't love this either -- as discussed in slack I was debating my options on how I wanted to handle this. What I _really_ wanted was a way to keep the method on the logging system typesafe so we could benefit from the existing ECS types, but I was weighing whether it was worth the complexity or would make performance worse.

But I think I found a way to refactor and arrive in a better place, while minimizing the work that needs to be done inside the `LoggingAdapter`. The end result is:

```ts
type GlobalContext = Record<string, unknown>; // flattened object with dot-separated paths as keys
loggingSystem.setGlobalContext(meta: Partial<LogMeta>);
loggingAdapter.updateGlobalContext(context: GlobalContext);
```

So the design still feels a little weird since `updateGlobalContext` is expecting a record in a special "flattened" format, but it saves the adapter from needing to do this work repeatedly, which feels like a fair balance. This also solved the issue of conflicting paths, which you had mentioned in another comment.